File: leetcode3/염혜정/3833. Count Dominant Indices.java
// nums[0] ~ nums[length -1] 합으로 평균을 구함
// O(n)

class Solution {
    public int dominantIndices(int[] nums) {
        if (nums.length == 1) return 0;

        int sum = 0;
        for (int i = 0; i<nums.length; i++) sum += nums[i];

        int cnt = 0;
        for (int i = 0; i<nums.length-1; i++) {
            sum -= nums[i];
            float avg = sum / (nums.length - i - 1);
            if (nums[i] > avg) cnt++;
        }
        return cnt;
    }
}

File: leetcode3/최원준/3833. Count Dominant Indices.py
#

'''
1. 아이디어 :
총합을 구하고 인덱스를 옮겨가며 차감 후 계산

2. 시간복잡도 :
    O(n + n)

3. 자료구조/알고리즘 :
-

'''
class Solution:
    def dominantIndices(self, nums: List[int]) -> int:
        n = len(nums)
        total = sum(nums)
        count = n

        ans = 0
        for i in range(n-1):
            total -= nums[i]
            count -= 1
            ans = ans + 1 if nums[i] > total/count else ans
        return ans
